src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package mate.academy.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private TicketDao ticketDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Optional<ShoppingCart> shoppingCartOptional
+                = shoppingCartDao.getByUser(user);
+        if (shoppingCartOptional.isPresent()) {
+            Ticket ticket = new Ticket();
+            ticket.setUser(user);
+            ticket.setMovieSession(movieSession);
+            ticketDao.add(ticket);
+            ShoppingCart shoppingCart = shoppingCartOptional.get();
+            List<Ticket> tickets = shoppingCart.getTickets();
+            tickets.add(ticket);
+            shoppingCart.setTickets(tickets);
+            shoppingCartDao.update(shoppingCart);
+            return;
+        }
+        throw new RuntimeException("Can't find shoppingCart by user: " + user);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        Optional<ShoppingCart> shoppingCartOptional
+                = shoppingCartDao.getByUser(user);
+        return shoppingCartOptional.orElseThrow(
+                () -> new RuntimeException("Can't find shoppingCart by user: " + user));
```

Comment:
```suggestion
       return shoppingCartDao.getByUser(user).get();
```
Do we have a reason for this? If we invoke the get() method, the result will be the same (ShopingCart or Exception)